pkg/k8s/client.go

```diff
@@ -68,6 +75,21 @@ func (c *KubeClient) GetPodName(podip string) (podns, poddname string, err error
 	return "", "", fmt.Errorf("match failed, ip:%s matching pods:%v", podip, podList)
 }
 
+func (c *KubeClient) getPodListWithTries(podip string, tries int) (*v1.PodList, error) {
+	podList, err := c.ClientSet.CoreV1().Pods("").List(metav1.ListOptions{
+		FieldSelector: "status.podIP==" + podip + ",status.phase==Running",
+	})
+
+	if err != nil || len(podList.Items) == 0 {
```

Comment:
nit: Do we need to retry unnecessarily if its a fatal error. Since there is max retry time, this is not a fatal issue, but was wondering if there is a way for us to distinguish real errors vs expected errors ?